includes/class-wc-api.php

```diff
@@ -40,6 +43,34 @@ public function __construct() {
 	}
 
 	/**
+	 * Send the response/close connection before shutdown where extra (slow)
+	 * logic, such as syncs, may be running.
+	 *
+	 * @since 3.1.2
+	 */
+	public function rest_send_reponse_before_shutdown() {
+		if ( ! defined( 'REST_REQUEST' ) || ! REST_REQUEST || headers_sent() ) {
+			return;
+		}
+
+		if ( session_id() ) {
+			session_write_close();
+		}
+
+		// This is the cleanest approuch when available and is used on nginx.
+		if ( is_callable( 'fastcgi_finish_request' ) ) {
+			fastcgi_finish_request();
+
+		// For apache, send some headers.
+		} else {
+			header( 'Content-Length: ' . ob_get_length() );
```

Comment:
You probably want to wrap these in a `headers_sent()` check, otherwise it'll throw errors if any output has already been sent.